plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/files/PropertyReload.java

```diff
@@ -0,0 +1,194 @@
+     * @return
+     */
+    public static Set<String> checkAndGetReloadBeanNames(DefaultListableBeanFactory beanFactory) {
+        Set<String> recreatedBeans = new HashSet<>();
+        // resolve constructor arguments
+        for (String beanName : beanFactory.getBeanDefinitionNames()) {
+            BeanDefinition beanDefinition = beanFactory.getBeanDefinition(beanName);
+            if (beanDefinition instanceof AnnotatedBeanDefinition) {
+                if (beanDefinition instanceof RootBeanDefinition) {
+                    RootBeanDefinition currentBeanDefinition = (RootBeanDefinition) beanDefinition;
+                    if (currentBeanDefinition.getFactoryMethodName() != null && currentBeanDefinition.getFactoryBeanName() != null) {
+                        Method method = currentBeanDefinition.getResolvedFactoryMethod();
+                        if (method == null) {
+                            Object factoryBean = beanFactory.getBean(currentBeanDefinition.getFactoryBeanName());
+                            Class factoryClass = ClassUtils.getUserClass(factoryBean.getClass());
+                            Method[] methods = ConstructorUtils.getCandidateMethods(factoryClass, currentBeanDefinition);
+                            for (Method m : methods) {
+                                if (!Modifier.isStatic(m.getModifiers()) && currentBeanDefinition.isFactoryMethod(m) &&
+                                        m.getParameterCount() != 0 && containValueAnnotation(m.getParameterAnnotations())) {
+                                    recreatedBeans.add(beanName);
+                                    break;
+                                }
+                            }
+                        } else if (method.getParameterCount() != 0) {
+                            // @Bean method contains @Value parameter
+                            if (containValueAnnotation(method.getParameterAnnotations())) {
+                                recreatedBeans.add(beanName);
+                            }
+                        }
+                    }
+                } else if (beanDefinition instanceof GenericBeanDefinition) {
+                    GenericBeanDefinition currentBeanDefinition = (GenericBeanDefinition) beanDefinition;
+                    if (BeanFactoryProcessor.checkNeedReload(beanFactory, currentBeanDefinition, beanName, constructors -> {
+                        for (Constructor constructor : constructors) {
+                            if (constructor.getParameterCount() != 0 && containValueAnnotation(constructor.getParameterAnnotations())) {
+                                return true;
+                            }
+                        }
+                        return false;
+                    })) {
+                        recreatedBeans.add(beanName);
+                    }
+                    // fixme factory method
+                } else if (beanDefinition instanceof AnnotatedGenericBeanDefinition) {
```

Comment:
This branch is useless since GenericBeanDefinition is a super class of AnnotatedBeanDefinition. Does this indicate this branch can be removed completely?

Reply:
the 'else if' branch still exists, it looks to me an useless branch since

```java
public class AnnotatedGenericBeanDefinition extends GenericBeanDefinition implements AnnotatedBeanDefinition 
```

Reply:
I will remove it for now.  I forget why I write the code.